hpy/debug/src/debug_internal.h

```diff
@@ -88,6 +88,10 @@ typedef struct DebugHandle {
     UHPy uh;
     long generation;
     bool is_closed;
+    // pointer to and size of any raw data associated with
+    // the lifetime of the handle:
+    void *associated_data;
+    HPy_ssize_t associated_data_size;
```

Comment:
this looks certainly good for this specific PR.
However, we are wasting some space even for handles which don't require these fields (which are most of them).
For the future, we might want to design a sort of "`DebugHandle` subclassing" to make it possible to create different kinds of handles with different kinds of extra data, maybe?

Reply:
Yes, I was concerned about this too. For now I don't see any other significantly better solution. I think the subclassing would not play well with the reusing of closed handles? In the future, in case we need to attach even more data to the debug handles, maybe we can use indirection with just one field in `DebugHandle` pointing to `DebugHandleData` or such and that would be allocated only if needed.

Reply:
yes, if we do "subclassing" we would need to maintain a separate queue of closed handle for each different kind (or at least for each different sizeof()).
`DebugHandleData` works in theory but I don't know how good it is in practice, because you pay the cost of the pointer + the hidden space cost of an extra malloc (but I don't know how much they are).

Another idea to keep into consideration is to use a `union` if/when we will need more kind of data.